content/en/blog/_posts/2023-04-11-volume-group-snapshot-alpha.md

```diff
@@ -0,0 +1,165 @@
+  source:
+     volumeGroupSnapshotHandle: group-snap-id
+  volumeGroupSnapshotRef:
+    kind: VolumeGroupSnapshot
+    name: pre-existing-group-snap1
+    namespace: demo-namespace
+```
+
+A VolumeGroupSnapshot object should be created to allow a user to use the group snapshot:
+```
+apiVersion: groupsnapshot.storage.k8s.io/v1alpha1
+kind: VolumeGroupSnapshot
+metadata:
+  name: pre-existing-group-snap1
+  namespace: demo-namespace
+spec:
+  snapshotContentName: pre-existing-group-snap-content1
+```
+
+Once these objects are created, the snapshot controller will bind them together, and set the field Ready (under Status) to True to indicate the group snapshot is ready to use.
+
+### How to use group snapshot for restore in Kubernetes
+At restore time, the user can request a new PersistentVolumeClaim to be created from a VolumeSnapshot object that is part of a VolumeGroupSnapshot. This will trigger provisioning of a new volume that is pre-populated with data from the specified snapshot. The user should repeat this until all volumes are created from all the snapshots that are part of a group snapshot.
+
+## As a storage vendor, how do I add support for group snapshots to my CSI driver?
+To implement the volume group snapshot feature, a CSI driver MUST:
+* Implement a new group controller service.
+* Implement group controller RPCs: `CreateVolumeGroupSnapshot`, `DeleteVolumeGroupSnapshot`, and `GetVolumeGroupSnapshot`.
+* Add group controller capability `CREATE_DELETE_GET_VOLUME_GROUP_SNAPSHOT`.
+
+See the [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md) and the [Kubernetes-CSI Driver Developer Guide](https://kubernetes-csi.github.io/docs/) for more details.
+
+Although Kubernetes poses as little prescriptive on the packaging and deployment of a CSI Volume Driver as possible, it provides a suggested mechanism to deploy a containerized CSI driver to simplify the process.
+
+As part of this recommended deployment process, the Kubernetes team provides a number of sidecar (helper) containers, including the [external-snapshotter sidecar container](https://kubernetes-csi.github.io/docs/external-snapshotter.html) which has been updated to support volume group snapshot.
+
+The external-snapshotter watches the Kubernetes API server for the `VolumeGroupSnapshotContent` object and triggers `CreateVolumeGroupSnapshot` and `DeleteVolumeGroupSnapshot` operations against a CSI endpoint.
+
+## What are the limitations?
+The alpha implementation of volume group snapshots for Kubernetes has the following limitations:
+* Restore is achieved by creating a new PVC from a VolumeSnapshot that is part of a VolumeGroupSnapshot.
+* Does not support reverting an existing PVC to an earlier state represented by a snapshot that is part of a group snapshot (only supports provisioning a new volume from a snapshot).
+* No application consistency guarantees beyond any guarantees provided by the storage system (e.g. crash consistency).
```

Comment:
Can you handle that yourself by using, eg, a sidecar to call `fsfreeze`?

Reply:
Yes